pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -432,7 +432,10 @@ func (s *DiscoveryServer) removeEdsCon(clusterName string, node string, connecti
 	if len(c.EdsClients) == 0 {
 		edsClusterMutex.Lock()
 		defer edsClusterMutex.Unlock()
-		adsLog.Infof("EDS: remove unused cluster node=%s cluster=%s all=%v", node, clusterName, edsClusters)
+		// This happens when a previously used cluster is no longer watched by any
+		// sidecar. It should not happen very often - normally all clusters are sent
+		// in CDS requests to all sidecars. It may happen if all connections are closed.
+		adsLog.Infof("EDS: remove unwatched cluster node=%s cluster=%s", node, clusterName)
```

Comment:
debug?

Reply:
I'm still not sure why it happens - it may indicate something is wrong. Normal case is if a service
is removed, but some abnormal cases can also cause it, so better to know.